src/VscPreviewExtension/media/buildTree.js

```diff
@@ -0,0 +1,319 @@
+      .on('click', textClick);
+
+    nodeEnter.append("title")
+      .text(function (d) { return getAltContent(d.name); });
+
+    // Transition nodes to their new position.
+    var nodeUpdate = node.transition()
+      .duration(durationTime)
+      .attr("transform", function (d) { return "translate(" + d.y + "," + d.x + ")"; });
+
+    nodeUpdate.select("circle")
+      .attr("r", 4.5)
+      .style("fill", function (d) { return d._children ? "lightsteelblue" : "white"; });
+
+    // Fade the text in
+    nodeUpdate.select("text")
+      .style("fill-opacity", 1);
+
+    // Transition exiting nodes to the parent's new position.
+    var nodeExit = node.exit().transition()
+      .duration(durationTime)
+      .attr("transform", function (d) { return "translate(" + source.y + "," + source.x + ")"; })
+      .remove();
+
+    nodeExit.select("circle")
+      .attr("r", 1e-6);
+
+    nodeExit.select("text")
+      .style("fill-opacity", 1e-6);
+
+    // Update the links
+    var link = vis.selectAll("path.link")
+      .data(tree.links(nodes), function (d) {
+        return d.target.id;
+      });
+
+    // Enter any new links at the parent's previous position.
+    link.enter().insert("path", "g")
+      .attr("class", "link")
+      .attr("d", function (d) {
+        var o = { x: source.x0, y: source.y0 };
+        return diagonal({ source: o, target: o });
+      });
+
+    // Transition links to their new position.
+    link.transition()
+      .duration(durationTime)
+      .attr("d", diagonal);
+
+    // Transition exiting nodes to the parent's new position.
+    link.exit().transition()
+      .duration(durationTime)
+      .attr("d", function (d) {
+        var o = { x: source.x, y: source.y };
+        return diagonal({ source: o, target: o });
+      })
+      .remove();
+
+    // Stash the old positions for transition.
+    nodes.forEach(function (d) {
+      d.x0 = d.x;
+      d.y0 = d.y;
+    });
+  }
+
+  // Function to center node when clicked so node will be tranlated to the center(depend on the depth)
+  function centerNode(source) {
+    scale = zoomListener.scale();
+    x = -source.y0;
+    y = -source.x0;
+    x = x * scale + width / 3 + source.depth * 100;   // Center of different level will be different
+    y = y * scale + height / 2;
+    d3.select('g').transition()
+      .duration(durationTime)
+      .attr("transform", "translate(" + x + "," + y + ")scale(" + scale + ")");
+    zoomListener.scale(scale)
+      .translate([x, y]);
+  }
+
+  // Toggle children on click.
+  function circleClick(d) {
+    if (d.children) {
+      d._children = d.children;
+      d.children = null;
+    } else {
+      d.children = d._children;
+      d._children = null;
+    }
+    update(d);
+    centerNode(d);
+  }
+
+  // Map to left editor when click the text
+  function textClick(d) {
+    $.get("http://localhost:" + port.toString() + "/MatchFromRightToLeft/" + getStartLineNumber(d.name) + "/" + getEndLineNumber(d.name) + "/");
```

Comment:
Yes!

---
In reply to: [80409738](https://github.com/dotnet/docfx/pull/693#discussion_r80409738) [](ancestors = 80409738)